tests/e2e_logging.py

```diff
@@ -1196,6 +1197,43 @@ def test_receipts(network, args):
     return network
 
 
+@reqs.description("Validate all receipts")
+@reqs.supports_methods("receipt", "log/private")
+@reqs.at_least_n_nodes(2)
+def test_all_receipts(network, args):
+    primary, _ = network.find_primary_and_any_backup()
+    cert_path = os.path.join(primary.common_dir, f"{primary.local_node_id}.pem")
+    with open(cert_path) as c:
+        node_cert = load_pem_x509_certificate(
+            c.read().encode("ascii"), default_backend()
+        )
+
+    with primary.client("user0") as c:
+        r = c.get("/app/commit")
+        max_view, seqno = [int(e) for e in r.body.json()["transaction_id"].split(".")]
+        view = 2
+        for s in random.sample(range(1, seqno - 1), 50):
```

Comment:
Doing them all is fairly expensive it turns out.

Reply:
A few things here:
- Why is this `seqno - 1`, does that mean we'll never sample the final signature?
- Can we replace `50` with `min(50, seqno)`, to avoid `ValueError: Sample larger than population or is negative`
- `random.sample` doesn't return sorted results, so the view-incrementing logic below is wrong (eg - we're fetching `42`, so count up to `5.42`, but then sample `35`, which is actually `3.35`)

Reply:
Yes!